DigitalLearningSolutions.Data/Services/RegistrationService.cs

```diff
@@ -471,9 +475,17 @@ DelegateRegistrationModel delegateRegistrationModel
             );
         }
 
-        private void ValidateCentreEmail(string centreEmail, int centreId)
+        private void ValidateCentreEmail(string centreEmail, int centreId, int? userId = null)

```

Comment:
I've renamed `userId`. I'm not super keen on moving the code to the RegistrationEmailValidator though, since a) it isn't currently called from any services, and b) its existing functions behave differently, i.e. adding errors to ModelState rather than throwing.

Reply:
Fair